docs/R-IOC/r10_tidyverse.md

```diff
@@ -0,0 +1,222 @@
+## 5                                           The Empire Strikes Back, Revenge of the Sith, Return of the Jedi, A New Hope, The Force Awakens
+## 6                                                                                     Attack of the Clones, Revenge of the Sith, A New Hope
+##                             vehicles                starships
+## 1 Snowspeeder, Imperial Speeder Bike X-wing, Imperial shuttle
+## 2                                                            
+## 3                                                            
+## 4                                             TIE Advanced x1
+## 5              Imperial Speeder Bike                         
+## 6          
+```
+
+!!! abstract "Take a break & Read"
+    For a detailled description of `tibbles`, please go read the [section 10](https://r4ds.had.co.nz/tibbles.html) 
+    of R4DS (O'Reilly Book on tidyverse).
+
+
+### Manipulating data
+
+Some of tidyverse packages such as `dplyr`, `tidyr` and `stringr` are used for manipulating 
+your data. You can add/remove columns or rows, filter or manipulating strings and tables. 
+
+!!! abstract "Take a break & Read"
+    To introduce the tidyverse, you can read the [section 3 Data Transform](https://r4ds.hadley.nz/data-transform) 
+    of R4DS 2e edition where you can be able to see basic commands for manipulating 
+    `data.frames` and `tibbles` thanks to `dplyr`.
+
+#### Pipe operator
+
+In the previous section, you must have notice some weird code : `|>`. It's called a 
+*pipe operator*. Tidyverse used this weird grammar to improve the code lisibility when
+you combine multiple functions to obtain one variable. 
+
+The following code lines are equivalent : 
+
+```
+# with the pipe operator
+starwars |> 
+  filter(species == "Droid") |>
+  head()
+
+
+#without the pipe operator
+head(filter(starwars, species == "Droid"))
+```
+
+But for those cases where you can combine several functions that may be more complicated, it can 
+really begin to be messy and difficult to read. You can imagine "russian dolls" where using
+the pipe is when they are not stack, it's the best way to know and see how many dolls we have and
+what they looks like.
+
+!!! abstract "Take a break & Read"
+    In order to be an expert of pipe operator, please go read again [section 3.4](https://r4ds.hadley.nz/data-transform#sec-the-pipe) 
+    but also [section 4.3](https://r4ds.hadley.nz/workflow-style#sec-pipes) of R4DS 2e edition that show you the best practices.
+
+#### Tidy data
+
+Using tibbles is not enough to make full use of the tidyverse's functionalities, you need to
+tidy your data. 
+Tidy a data.frame or a tibble is a manipulation of the columns in order to obtain one column = 
+one variable and one row = one observation.
+
+For instance here is a the difference between tidy and untidy data : 
+
+```
+# Untidy data
+untidy_df <- data.frame(celltype = c("T-cells", "B-cells", "Macrophages", "Endothelial cells", 
```

Comment:
For me, it is a tidy data which we just need to transpose it to meet you criteria in the above paragraph.
What `pivot_longer` does is just re-arrange data in a different form.